pycode/memilio-epidata/memilio/epidata/modifyDataframeSeries.py

```diff
@@ -141,6 +173,9 @@ def impute_and_reduce_df(
 
             # compute 'moving average'-days moving average
             if moving_average > 0:
+                # extract subframe to prevent unnecessary computation
+                df_local_new = extract_subframe_based_on_dates(
+                    df_local_new, min_date_to_use, max_date_to_use)
```

Comment:
Why does this extraction only happen for ma>0?

Reply:
the `date_to_use` is ma/2 shifted for computing the moving average. Afterwards the resulting dataframe is extracted on the given min_date/max_date (independent of moving_average or impute_dates)